lasy/utils/laser_utils.py

```diff
@@ -1285,3 +1283,146 @@ def get_propation_angle(dim, grid, k0):
+
+
+def get_spectral_phase(grid, dim, omega0, method="sum"):
+    r"""
+    Calculate the spectral phase of a pulse in a given grid.
+
+    Depending on the chosen calculation method, the spectral phase can be calculated in two different ways.
+
+    If `method==sum` (default), the spectral field of the laser is spatially integrated before extracting the phase, i.e.
+
+    .. math::
+
+        \varphi(\omega) = \arg\left( \int E(x,y,\omega) dx dy \right)
+
+    If `method==on-axis`, the on-axis spectral phase is calculated:
+
+    .. math::
+
+        \varphi(\omega) = \arg\left( E(x=0,y=0,\omega) \right)
+
+    Parameters
+    ----------
+    grid : Grid
+        The grid with the field to analyze.
+
+    dim : string
+        Dimensionality of the array. Options are:
+
+        - ``'xyt'``: The laser pulse is represented on a 3D grid:
+                    Cartesian (x,y) transversely, and temporal (t) longitudinally.
+        - ``'rt'`` : The laser pulse is represented on a 2D grid:
+                    Cylindrical (r) transversely, and temporal (t) longitudinally.
+
+    omega0 : float
+        Central angular frequency of the field
+
+    method : string, optional
+        Method of calculating the spectral phase. Options are:
+
+        - ``'sum'``: Calculates the spectral phase of the spatially summed field (default).
+        - ``'on-axis'``: Calculates the on-axis spectral phase.
+
+    Returns
+    -------
+    phase: ndarray of floats (1D)
+        Spectral phase of the pulse in the specified units and calculation method (in rad/s)
+
+    omega: ndarray of floats (1D)
+        Angular freuqencies at which the phase is defined (in rad/s)
+
+    """
+    # Field must be envelope
+    assert grid.is_envelope
+
+    # get the spectral field
+    field_spectral = grid.get_spectral_field()
+    field_spectral = np.fft.fftshift(field_spectral, axes=-1)
+
+    # if method=='on-axis' get the on-axis field envelope, and calculate its phase
+    if method == "on-axis":
+        if dim == "xyt":
+            Nx = grid.npoints[0]
+            Ny = grid.npoints[1]
+            phase = np.angle(field_spectral[Nx // 2, Ny // 2, :])
+        else:  # dim=='rt'
+            phase = np.angle(field_spectral[0, 0, :])
+
+    # if method=='sum' integrate the field spatially before getting the phase from it
+    else:  # method='sum'
+        # grid cell volume required for integration
+        dV = get_grid_cell_volume(grid, dim)
+
+        if dim == "xyt":
+            summed_field = np.sum(field_spectral * dV, axis=(0, 1))
+        else:  # dim=='rt'
+            summed_field = np.sum(field_spectral * dV[None, :, None], axis=(0, 1))
+
+        phase = np.angle(summed_field)
+
+    # unwrap the phase
+    phase = np.unwrap(phase)
+
+    # create omega array (angular frequencies)
+    omega = np.fft.fftshift(
+        2 * np.pi * np.fft.fftfreq(grid.npoints[-1], grid.dx[-1]), axes=-1
+    )
+
+    if omega0:
+        omega += omega0
+
```

Comment:
`omega0` isn't optional, is it? I think this should always be done.
```suggestion
    omega += omega0

```